### DISCUSSION TITLE
How detekt reports can be aggregated if maven used?

### DISCUSSION THREAD

**Discussion by pavlo-noumena** [detekt/detekt#5688] [Q&A] [answered]
How xml\html versions of detekt reports can be aggregated with maven

**Comment by schalkms** [comment, marked as answer]
Detekt supports merging XML and SARIF out of the box via Gradle plugin. See [here](https://detekt.dev/docs/introduction/reporting/#merging-reports) for more details.

Do you use the 3rd party detekt plugin for Maven by ozsie?

**Reply by pavlo-noumena** [reply]
Yes, I'm interested how maven-based projects can aggregate detekt xml reports.
any recommendations ?

**Reply by schalkms** [reply]
There's no official support in detekt to aggregate XML reports via Maven.
The recommendation is to check the 3rd party plugin by oszie whether it's supported there.
Alternatively, you could write a custom Maven plugin, which aggregates all the reports for each Maven module.
